File: common/src/main/java/generations/gg/generations/core/generationscore/network/packets/S2CUnlockReloadPacket.java
package generations.gg.generations.core.generationscore.network.packets;

import net.minecraft.network.FriendlyByteBuf;
import net.minecraft.resources.ResourceLocation;;

import static generations.gg.generations.core.generationscore.GenerationsCore.id;

public class S2CUnlockReloadPacket implements GenerationsNetworkPacket<S2CUnlockReloadPacket> {
    public static ResourceLocation ID = id("unlock_reload");

    public static S2CUnlockReloadPacket decode(FriendlyByteBuf buffer) {
        return new S2CUnlockReloadPacket();
    }

    @Override
    public ResourceLocation getId() {
        return ID;
    }

    public void encode(FriendlyByteBuf buffer) {}

}

File: src/common/main/kotlin/generations/gg/generations/core/generationscore/common/network/packets/S2CUnlockReloadPacket.kt
package generations.gg.generations.core.generationscore.common.network.packets

import com.cobblemon.mod.common.api.net.NetworkPacket
import generations.gg.generations.core.generationscore.common.GenerationsCore
import net.minecraft.network.FriendlyByteBuf
import net.minecraft.network.RegistryFriendlyByteBuf
import net.minecraft.resources.ResourceLocation

class S2CUnlockReloadPacket : NetworkPacket<S2CUnlockReloadPacket> {
    override val id: ResourceLocation = ID

    override fun encode(buffer: RegistryFriendlyByteBuf) {}

    companion object {
        var ID: ResourceLocation = GenerationsCore.id("unlock_reload")

        fun decode(buffer: RegistryFriendlyByteBuf): S2CUnlockReloadPacket {
            return S2CUnlockReloadPacket()
        }
    }
}